### PR TITLE
(BIDS-1911) Fix sync period handling for exiting validators

### PR DESCRIPTION
**Pull request by D13ce**

### CODE REVIEW

File: handlers/api.go [hunk 1]
```diff
@@ -1075,15 +1075,13 @@ func getExpectedSyncCommitteeSlots(validators []uint64, epoch uint64) (expectedS
...
+		if vi.ExitEpoch > epoch || vi.ExitEpoch == noEpoch {
 			lastEpoch = epoch
 		}
 
-		{
-			spoe := utils.SyncPeriodOfEpoch(lastEpoch)
-			if _, found := periodInfoMap[spoe]; !found {
-				return 0, fmt.Errorf("required period not found: SyncPeriodOfEpoch(%v) = '%v'", lastEpoch, spoe)
-			}
+		spoe := utils.SyncPeriodOfEpoch(lastEpoch)
```
Comment (recy21) [inline handlers/api.go:1082]: If you remove the curly braces, spoe will be available outside the scope, for no reason?

Comment (D13ce) [reply handlers/api.go:1082]: In this very case we are talking about a harmless number. Protecting it with its own scope seems excessive to me.

Personally I believe that perfectly scoping every single variable leads to unreadable code as more and more intending is added.

Comment (recy21) [reply handlers/api.go:1082]: @D13ce in that case it should be:
`
		spoe := utils.SyncPeriodOfEpoch(lastEpoch)
		if entry, found := periodInfoMap[spoe]; found {
			expectedCommitties += entry - periodInfoMap[utils.SyncPeriodOfEpoch(vi.ActivationEpoch)]	
		} else {
			return 0, fmt.Errorf("required period not found: SyncPeriodOfEpoch(%v) = '%v'", lastEpoch, spoe)
		}
	}
`

Comment (D13ce) [reply handlers/api.go:1082]: You are right, `spoe` should be used more = Done.

Comment (D13ce) [reply handlers/api.go]: Okay, thanks for making me look again. I have improved the code in general now and it should now even be more secure.

**Review by Eisei24 — APPROVED** [review]
lgtm

**Pull request merged**